solutions/usaco-553.mdx

```diff
@@ -0,0 +1,97 @@
+---
+id: usaco-553
+source: USACO Gold 2015 Open
+title: Palindromic Paths
+author: Maggie Liu
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_palpath_gold.html)
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N^3)$
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <cstdio>
+#include <string>
+#include <algorithm>
+using namespace std;
+
+int main()
+{
+	freopen("palpath.in", "r", stdin);
+	freopen("palpath.out", "w", stdout);
+	int n;
+	cin >> n;
+	string grid[500];
+	for (int i = 0; i < n; i++)
+	{
+		cin >> grid[i];
+	}
+	long long prev[500][500]{};
```

Comment:
1. plz use vectors
2. no magic numbers, maybe have a `MAX_N` or something

Reply:
1. why vectors?
2. sure

Reply:
~~arrays make me uncomfortable~~
if you're gonna use them make them global like everyone else does

Reply:
but there's nothing wrong with local arrays, right? is there a problem with memory?

Reply:
i,,,honestly don't know

Reply:
ok so this is fine